plugins/action/query_graphql.py

```diff
@@ -90,8 +97,14 @@ def nautobot_action_graphql(args):
 
     # Good result, return it
     if isinstance(nautobot_response, pynautobot.core.graphql.GraphQLRecord):
-        # Assign the data of a good result to the response
-        results["data"] = nautobot_response.json.get("data")
+        # If populate_root is set, add to ansible_facts which will set to the root of
+        # the data structure, e.g. hostvars[inventory_hostname]
+        if args.get("populate_root"):
+            results["ansible_facts"] = nautobot_response.json.get("data")
+        else:
```

Comment:
I don't think we need the else here as we should be returning the data regardless. The flag should just set the facts or not.

Reply:
I had that at first, it seemed like duplicate, what say everyone else?

Reply:
Let's go ahead and remove the else, remove the indent for the results. Then the data key will be populated regardless, whether it has data or it is `None`.